### ISSUE TITLE
Show all tests in VSCode's Testing sidebar

### ISSUE THREAD

**Issue by Juice10** [Shopify/ruby-lsp#1155]
### I have checked that this feature is not already implemented

- [X] This feature does not exist

### Use case

Currently I'm using jest, vitest and mocha (Test Explorer) extensions as well as Ruby LSP.
Apart from Ruby LSP, these list all of the tests available in the project by default.
I understand that we're not doing that for performance/navigation reasons on large projects. But for projects where ruby/rails tests are still manageable it would be nice to have the option to display them all in the sidebar and keep the defacto default behaviour people have come to expect from other extensions.

### Description

I would like to see all available ruby tests in the vscode sidebar.

### Implementation

Since we are disabling all tests for large projects for navigation/performance reasons, it would be nice if we could configure if we show all tests (or don't). Personally I think we should show all tests by default as most other test explorers I've worked with do that. (although I acknowledge it might be a process before that is feasible for ruby-lsp).

It would be nice if we could respect rails' `parallelize` configuration so each test doesn't fire off its own process.
```ruby
module ActiveSupport
  class TestCase
    # Run tests in parallel with specified workers
    parallelize(workers: :number_of_processors)
  end
end
```

**Comment by egiurleo** [comment]
This has been accomplished in our recent test discovery project! See #3170.
